### ISSUE TITLE
Attempting to unparent a Nested Prefab while editing it may crash the Editor

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#4862]
**Describe the bug**
Attempting to unparent a Nested Prefab while being in its Focus Mode may crash the Editor. Please note that the reproduction rate of the issue is not 100% and sometimes requires a couple of attempts to occur. 
If the crash does not occur, the following error is printed in the Console:

`[Error] (Prefab) - An empty list of entities and prefab instances were retrieved from the selected entities`

Along with the following Asserts when reproducing the issue (they can all be found in the attached Asserts.log):
```
(System) - Trace::Assert
 C:\Builds\o3de\Code\Framework\AzCore\AzCore/std/optional.h(119): (7324) 'class std::reference_wrapper<class AZ::Entity> &__cdecl AZStd::Internal::OptionalImplBase<class std::reference_wrapper<class AZ::Entity> >::value(void) &'
(System) - Optional does not have a value
```
```
(System) - Trace::Assert
 C:/Builds/o3de/Code/Framework/AzCore/AzCore/Memory/SystemAllocator.cpp(261): (7324) 'void *__cdecl AZ::SystemAllocator::Allocate(unsigned __int64,unsigned __int64,int,const char *,const char *,int,unsigned int)'
(System) - SystemAllocator: Failed to allocate 1853657248 bytes aligned on 1 (flags: 0x00000000) AZStd::allocator : C:/Builds/o3de/Code/Framework/AzCore/AzCore/std/allocator.cpp (21)!
```
```
(System) - Trace::Assert
 C:\Builds\o3de\Code\Framework\AzCore\AzCore/std/string/string.h(1715): (7324) 'void __cdecl AZStd::basic_string<char,struct AZStd::char_traits<char>,class AZStd::allocator>::copy(unsigned __int64,unsigned __int64)'
<12:20:54> (System) - AZStd::string allocation failed!
```

Please note that the call stack is identical to the one in #4858, but the repro steps are different enough to consider these as two separate issues.

Please refer to the attached video and log files for more details.

**Steps to reproduce**
1. Open any Level.
2. Create an entity and make a Prefab out of it.
3. Double click the Prefab (or RMB -> Edit Prefab).
4. RMB the Entity Outliner and select Create Entity.
5. Make a Prefab out of the entity created in the previous step.
6. Double click the Nested Prefab (or RMB -> Edit Prefab).
7. With the Nested Prefab selected click the _X_ button on the _Parent entity_ field in Entity Inspector.
8. (optional) Repeat step 7 until the crash occurs.

**Expected behavior**
The Editor remains stable.

**Actual behavior**
The Editor crashes or an error and multiple Asserts occur.

**Call Stack Trace**
```
 7) 00007FFCD77FC347 (KERNELBASE) : UnhandledExceptionFilter
 6) 00007FFCDA0151B0 (ntdll) : memset
 5) 00007FFCD9FFC766 (ntdll) : _C_specific_handler
 4) 00007FFCDA0120CF (ntdll) : _chkstk
 3) 00007FFCD9FC1454 (ntdll) : RtlRaiseException
 2) 00007FFCDA010BFE (ntdll) : KiUserExceptionDispatcher
 1) D:\Git\o3de\Code\Framework\AzCore\AzCore\IO\Path\Path.inl (837) : AZ::IO::BasicPath<AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> >::operator/=
```

**Assets required**
N/A

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/138264561-ded43615-abcc-42a3-a371-696d97330d20.mp4

**Found in Branch**
Development (74d7405).

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU: AMD Ryzen 5 3600
 - GPU: Nvidia RTX 2060 SUPER
 - Memory: 16GB

**Attachments**
[Editor.log](https://github.com/o3de/o3de/files/7388439/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7388440/error.log)
[error.zip](https://github.com/o3de/o3de/files/7388441/error.zip)
[Asserts.log](https://github.com/o3de/o3de/files/7388447/Asserts.log)

**Comment by AMZN-daimini** [comment]
Hey Tomasz,

there is an incoming change that will hide the transform component from the container entity of the prefab that is being edited, since the operation you describe isn't valid. I'll attach this issue to that task so that it can be verified when that change is in.

Thank you :D

**Comment by AMZN-daimini** [comment]
Issue can no longer be reproduced as instructed (transform card is not visible in the Inspector when selecting the container entity of the focused prefab instance)

**Comment by LB-TomaszSerwanski** [comment]
Issue verified as fixed in the latest Development (bc74b3b).
